addons/web/controllers/main.py

```diff
@@ -2148,13 +2148,12 @@ def report_download(self, data, token, context=None):
             else:
                 return
         except Exception as e:
-            se = _serialize_exception(e)
             error = {
-                'code': 200,
+                'code': 500,
                 'message': "Odoo Server Error",
-                'data': se
+                'data': _serialize_exception(e)
             }
-            return request.make_response(html_escape(json.dumps(error)))
+            raise werkzeug.exceptions.InternalServerError(json.dumps(error)) from e
```

Comment:
Same as above, restore `'code': 200`, leave the `se` variable, add the missing content-type header